pkg/aggregator/check_sampler.go

```diff
@@ -37,7 +50,67 @@ func (cs *CheckSampler) addSample(metricSample *metrics.MetricSample) {
 	}
 }
 
-func (cs *CheckSampler) commit(timestamp float64) {
+func (cs *CheckSampler) newSketchSeries(ck ckey.ContextKey, points []metrics.SketchPoint) metrics.SketchSeries {
+	ctx := cs.contextResolver.contextsByKey[ck]
+	ss := metrics.SketchSeries{
+		Name: ctx.Name,
+		Tags: ctx.Tags,
+		Host: ctx.Host,
+		// Interval: TODO: investigate
+		Points:     points,
+		ContextKey: ck,
+	}
+
+	return ss
+}
+
+func (cs *CheckSampler) addBucket(bucket *metrics.HistogramBucket) {
+	if bucket.Value < 0 {
+		log.Warnf("Negative bucket value %d for metric %s discarding", bucket.Value, bucket.Name)
+		return
+	}
+	if bucket.Value == 0 {
+		// noop
+		return
+	}
+	contextKey := cs.contextResolver.trackContext(bucket, bucket.Timestamp)
+
+	// if we already saw the bucket we only send the delta
+	lastBucketValue, bucketFound := cs.lastBucketValue[contextKey]
+	rawValue := bucket.Value
+	if bucketFound {
+		cs.lastSeenBucket[contextKey] = time.Now()
+		bucket.Value = rawValue - lastBucketValue
+	}
+	cs.lastBucketValue[contextKey] = rawValue
+	cs.lastSeenBucket[contextKey] = time.Now()
+
+	// simple linear interpolation, TODO: optimize
+	if math.IsInf(bucket.UpperBound, 1) {
+		// Arbitrarily double the lower bucket value for interpolation over infinity bucket
```

Comment:
This is likely going to require some documentation, is there any guideline or standard practice around this in the community?

Reply:
I wasn't able to find any guideline expect what is currently done in the `histogram_quantile` function that requires the infinity bucket but will return the closest bucket if the quantile falls into it:
> if the quantile falls into the highest bucket, the upper bound of the 2nd highest bucket is returned

ref https://github.com/prometheus/prometheus/blob/41151ca8dc069448515f48893b8631b9a3ad8df8/promql/quantile.go#L49-L70

Reply:
Ah. That's confusing and unfortunate, but at least it's deterministic. I'll never understand why they don't just add a max metric. But anyway, for now we should mimc this I guess?

Reply:
@jbarciauskas I'm not sure how to emulate this behavior with the sketch, don't try to interpolate anything with the infinity bucket? (would mess up the `summary`)
Or interpolate everything close/at the value of "the upper bound of the 2nd highest" ? (close to what I'm currently doing)

Reply:
Yeah, I mean report the count of values in the infinity bucket as the last defined bucket value, essentially as if they were all = max. Either choice (2nd to last bucket = max or 2*2nd to last bucket = max) is synthesizing data that doesn't exist. The first one means that query results should be similar at least.

Reply:
just changed the logic to insert everything at the `lower_bound` of the infinity bucket

Reply:
Choosing the `lower_bound` makes sense to me, but this needs to be very loudly documented since `max` is ~always going to be wrong in fairly surprising ways.